src/julia.h

```diff
@@ -397,6 +397,7 @@ typedef struct {
     unsigned exportp:1;
     unsigned imported:1;
     unsigned deprecated:1;
+    jl_sym_t *deprecation_preferred_name;  // non-NULL when new type incompatible with old
```

Comment:
I just have a blanket objection to changing core data structures in order to get a slightly more useful warning message in one case.

It looks like some packages that dispatch on `::FloatRange` also access its fields, in which case changing the argument type to `::StepRangeLen` doesn't help. So deprecating `FloatRange` to `FloatRangeDeprecated`, and explaining elsewhere how to fully migrate, is about the best we can do.